sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -2889,5 +2889,129 @@ def math_pipeline():
                 add(nums=dsl.Collected(t.output))
 
 
+class TestValidIgnoreUpstreamTaskSyntax(unittest.TestCase):
+
+    def test_basic_permitted(self):
+
+        @dsl.component
+        def fail_op(message: str) -> str:
+            """Fails."""
+            import sys
+            print(message)
+            sys.exit(1)
+            return message
+
+        @dsl.component
+        def print_op(message: str = 'default'):
+            """Prints a message."""
+            print(message)
+
+        @dsl.pipeline()
+        def my_pipeline(sample_input1: str = 'message'):
+            task = fail_op(message=sample_input1)
+            clean_up_task = print_op(
+                message=task.output).ignore_upstream_failure()
+
+    def test_clean_up_on_wrapped_pipeline_permitted(self):
+
+        @dsl.component
+        def fail_op(message: str = 'msg') -> str:
+            """Fails."""
+            import sys
+            print(msg)
+            sys.exit(1)
+            return input_string
+
+        @dsl.component
+        def print_op(message: str = 'msg'):
+            """Prints a message."""
+            print(msg)
```

Comment:
Should be `message`. Same with `message` and `input_string` above.

Reply:
For this comment (and the above about undefined variables): while this doesn't affect the correctness of the test case or throw an exception, it has the potential to be misleading. In general, the test cases should introduce nothing unique other than the specific behavior being tested.